src/Uno.UWP/Devices/Sensors/Helpers/SensorHelpers.iOS.cs

```diff
@@ -6,12 +6,49 @@ namespace Uno.Devices.Sensors.Helpers
 {
 	internal static class SensorHelpers
 	{
+		private static readonly DateTimeOffset NSDateConversionStart =
+			new DateTimeOffset(2001, 1, 1, 0, 0, 0, TimeSpan.Zero);
+
 		public static DateTimeOffset TimestampToDateTimeOffset(double timestamp)
 		{
 			var bootTime = NSDate.FromTimeIntervalSinceNow(-NSProcessInfo.ProcessInfo.SystemUptime);
 			var date = (DateTime)bootTime.AddSeconds(timestamp);
 			return new DateTimeOffset(date);
 		}
+
+		public static DateTimeOffset NSDateToDateTimeOffset(NSDate nsDate)
```

Comment:
There are already similar methods in `NSDateExtensions`, do they work for your purpose?

Reply:
Unfortunately they are in `Uno.UI`, so I can't use those...

Reply:
Ah. Do they work, though? If so just move them. (Or move the implementation and redirect the `Uno.UI` methods for binary compatibility)

Reply:
Just realized they are actually a bit different - the `NSDateExtensions` work with `DateTime` while Sensors require `DateTimeOffset` instead, so it is not exactly the same and would require additional methods anyway.